site/component/manifests/wordpress.pp

```diff
@@ -0,0 +1,37 @@
+class component::wordpress (
+  $path             = hiera('path', '/var/www/app_name'),
+  $vhost            = hiera('vhost', 'app-name.dev'),
+  $vhost_port       = 80,
+  $ssl              = false,
+  $mutisite         = false
+) {
+  if $ssl {
+    exec {'create_self_signed_sslcert':
+      command => "openssl req -newkey rsa:2048 -nodes -keyout ${vhost}.key  -x509 -days 365 -out ${vhost}.crt -subj '/CN=${vhost}'",
```

Comment:
Please include functionality for not creating the keys but take it from the source root in `/vagrant`'.